src/analyses/mutexTypeAnalysis.ml

```diff
@@ -6,6 +6,13 @@ open Analyses
 module MAttr = ValueDomain.MutexAttr
 module LF = LibraryFunctions
 
+(* Removing indexes here avoids complicated lookups inside the map for a varinfo, at the price that different types of mutexes in arrays are not dinstinguished *)
+let rec offs_no_index o =
+  match o with
+  | `NoOffset -> `NoOffset
+  | `Field (f,o) -> `Field (f, offs_no_index o)
+  | `Index (i,o) -> `Index (Lval.any_index_exp, offs_no_index o)
```

Comment:
Isn't `MutexAnalysis.Spec.conv_offset_inv` already this, but slightly better?

Reply:
The comment refers to an earlier version, we now use `Lval.OffsetNoIdx.of_offs` everywhere.